cfpb/sbl-frontend
# fix(auth): update auth packages to prevent 403s

We've been experiencing flaky auth issues during the bug bashes: the `automaticSilentRenew` [feature](https://authts.github.io/oidc-client-ts/interfaces/UserManagerSettings.html#automaticSilentRenew) hasn't been working as advertised: it's supposed to attempt an automatic refresh [using an iframe](https://github.com/authts/oidc-client-ts/blob/066a8bcb189159d87b570a0ca052c139c2959d44/docs/protocols/silent-refresh-token-in-iframe-flow.md?plain=1#L19) to not force a re-render, but that wasn't happening. 

I tried [just calling the silent refresh directly](https://github.com/cfpb/sbl-frontend/pull/697) so I wouldn't have to update these libraries this close to launch, but that was also kind of a bust. So here we are: I updated the packages and now it works great?

I've updated the testing instructions so it's easy to see if it's working or not. I'll also push this to the preview environment, so it's easier to see if it's actually working.

## Changes

- updated the `oidc-client-ts` and `react-oidc-context` libraries

## How to test this PR

1. Navigate to the [regtech realm settings for tokens](http://localhost:8880/admin/master/console/#/regtech/realm-settings/tokens). Change the Access Token Lifespan to 65 seconds. (oidc-client-ts [tries to renew a token one minute](https://github.com/authts/oidc-client-ts/blob/0d9dd56/src/UserManagerSettings.ts#L52C117-L52C209) before the expiration time, so this will make the frontend seek a new token every five seconds)
![Screenshot 2024-06-19 at 1 51 42 PM](https://github.com/cfpb/sbl-frontend/assets/19983248/c339cb1c-bc20-4ccc-9903-e5ecf84b318b)

3. Go to the upload page and try to upload a large file with your network tab open.
4. Notice that the upload page does not refresh or re-render, but notice in the network tab successfully updates the token every 5 seconds.
5. You can also test that nothing is re-rendering by going to point of contact, making a change to the form, and seeing that your changed content doesn't change when the token is renewed.

## Screenshots

![Screenshot 2024-06-19 at 1 55 59 PM](https://github.com/cfpb/sbl-frontend/assets/19983248/30c1bc78-0398-475a-82bd-90f3d313a2b5)
